### PR TITLE
FRW-1893: Dynamic Data Exchange API

### PR DESCRIPTION
**Pull request by olhalivitchuk**
## PR Description

https://spryker.atlassian.net/browse/FRW-1893

- Introduced Integration Guide for Dynamic Data Exchange API.
- Introduced How-to configure Dynamic Data Exchange API guide.
- Introduced How-to send requests and operate with the data in Dynamic Data Exchange API. 

## Checklist
- [ ] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: docs/scos/dev/glue-api-guides/202304.0/dynamic-data-exchange-api/how-to-guides/how-to-send-request-in-dynamic-data-exchange-api.md [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+You will receive the following response which does not include `iso3_code` because the value was not set:
+
+```json
+[
+    {
+        "iso2_code": "WA",
+        "name": "FOO",
+        "id_country": 259
+    }
+]
```
Comment (asmarovydlo) [inline docs/scos/dev/glue-api-guides/202304.0/dynamic-data-exchange-api/how-to-guides/how-to-send-request-in-dynamic-data-exchange-api.md:225]: This is not the way how it should work.
All the readable fields should be returned in the response.
If the field is `isCreatable=false`, but it is provided in the request => 400 error should be returned, with a descriptive message.

File: docs/scos/dev/glue-api-guides/202304.0/dynamic-data-exchange-api/how-to-guides/how-to-send-request-in-dynamic-data-exchange-api.md [hunk 2]
```diff
@@ -0,0 +1,457 @@
...
+}
+```
+
+{% endinfo_block %}
+
+{% info_block infoBox %}
+
+Note that all fields included in the request payload are marked as `isEditable: true` in the configuration.
+Therefore, the API will update the found record with all the provided fields.
+However, if any of the fields are configured as `isEditable: false` it won't be updated.
```
Comment (asmarovydlo) [inline docs/scos/dev/glue-api-guides/202304.0/dynamic-data-exchange-api/how-to-guides/how-to-send-request-in-dynamic-data-exchange-api.md:321]: If any of those fields are not editable, but we are trying to edit them, we should receive 400 response with a descriptive message.
Please verify the implementation, and update the documentation.

File: docs/scos/dev/glue-api-guides/202304.0/dynamic-data-exchange-api/how-to-guides/how-to-send-request-in-dynamic-data-exchange-api.md [hunk 3]
```diff
@@ -0,0 +1,457 @@
...
+  }
+]
+```
+
+{% endinfo_block %}
+
+{% info_block infoBox %}
+
+Similarly to the `POST` request, it is important to consider database-specific configurations when sending a `PATCH` request. 
+For example, modifying the ID of the entity or providing duplicate values for unique columns, etc will result in an error.
```
Comment (asmarovydlo) [inline docs/scos/dev/glue-api-guides/202304.0/dynamic-data-exchange-api/how-to-guides/how-to-send-request-in-dynamic-data-exchange-api.md:370]: Please see the comment above, it should not result in an error.

**Review by asmarovydlo — CHANGES_REQUESTED** [review]

File: docs/scos/dev/glue-api-guides/202304.0/dynamic-data-api/how-to-guides/how-to-configure-dynamic-data-api.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+      "isEditable": true,
+      "isCreatable": true,
+      "validation": {
+        "isRequired": true,
+        "maxLength": 2,
+        "minLength": 2
+      }
+    }
+  ]
+}
```
Comment (asmarovydlo) [inline docs/scos/dev/glue-api-guides/202304.0/dynamic-data-api/how-to-guides/how-to-configure-dynamic-data-api.md:53]: https://github.com/spryker/spryker-docs/pull/1944#discussion_r1247915549 is not fixed.
The identifier field should also be defined in the `fields` array.

File: docs/scos/dev/feature-integration-guides/202304.0/glue-api/dynamic-data-api/dynamic-data-api-integration.md [hunk 5]
```diff
@@ -0,0 +1,363 @@
...
+### 2) Set up the configuration
+
+1. Move the following commands into `setup-glue` section after `demodata` step:
+
+**config/install/docker.yml**
+
+```yaml
+setup-glue:
+
+    ...
```
Comment (asmarovydlo) [inline docs/scos/dev/feature-integration-guides/202304.0/glue-api/dynamic-data-api/dynamic-data-api-integration.md:59]:
```suggestion
```

File: docs/scos/dev/glue-api-guides/202304.0/dynamic-data-api/how-to-guides/how-to-configure-dynamic-data-api.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+It is recommended to set `isEditable` and `isCreatable` to `false` for fields that serve as identifiers or keys, ensuring their immutability and preserving the integrity of the data model.
+
+{% endinfo_block %}
+
+{% info_block infoBox %}
+
+Configuring the definition for the field responsible for the identifier keep in mind that it JSON accepts numbers within 
+the range of a 64-bit signed integer (-9,223,372,036,854,775,808 to 9,223,372,036,854,775,807). However, if there is a need 
+to use identifiers outside this range or if the person providing the configuration anticipates larger values, the field can be set as a 
+string type instead.
```
Comment (asmarovydlo) [inline docs/scos/dev/glue-api-guides/202304.0/dynamic-data-api/how-to-guides/how-to-configure-dynamic-data-api.md:83]: 1. It is applicable not only to identifiers but to any bigint columns.
2. It is not JSON related limitation, see https://www.w3schools.com/php/php_numbers.asp#:~:text=PHP%20Integers&text=An%20integer%20data%20type%20is,9223372036854775807%20in%2064%20bit%20systems. for 32b systems.

File: docs/scos/dev/glue-api-guides/202304.0/dynamic-data-api/how-to-guides/how-to-send-request-in-dynamic-data-api.md [hunk 7]
```diff
@@ -0,0 +1,456 @@
...
+
+The response payload includes all the specified fields from the request body, along with the id of the newly created entity.
+
+{% endinfo_block %}
+
+{% info_block infoBox %}
+
+Note that all fields included in the request payload are marked as `isCreatable: true` in the configuration. 
+Therefore, the API will create a new record with all the provided fields. 
+However, if any of the fields are configured as `isCreatable: false` it won't be created.
```
Comment (asmarovydlo) [inline docs/scos/dev/glue-api-guides/202304.0/dynamic-data-api/how-to-guides/how-to-send-request-in-dynamic-data-api.md:213]: It is not always true: if it is `isCreatable: false`, but not required, the row will be created.

File: docs/scos/dev/feature-integration-guides/202304.0/glue-api/dynamic-data-api/dynamic-data-api-integration.md [hunk 8]
```diff
@@ -0,0 +1,363 @@
...
+dynamic_entity.validation.persistence_failed,"Das Speichern der Daten ist fehlgeschlagen. Bitte überprüfen Sie die bereitgestellten Daten und versuchen Sie es erneut.",de_DE
+dynamic_entity.validation.persistence_failed_duplicate_entry,"Failed to persist the data. Please verify the provided data and try again. Entry is duplicated.",en_US
+dynamic_entity.validation.persistence_failed_duplicate_entry,"Das Speichern der Daten ist fehlgeschlagen. Bitte überprüfen Sie die bereitgestellten Daten und versuchen Sie es erneut. Eintrag is doppelt vorhanden.",de_DE
+dynamic_entity.validation.entity_does_not_exist,"The entity could not be found in the database.",en_US
+dynamic_entity.validation.entity_does_not_exist,"Die Entität konnte in der Datenbank nicht gefunden werden.",de_DE
+dynamic_entity.validation.required_data_is_missing_or_forbidden,"Some required data is missing or provided data cannot be modified. Please verify the request and try again.",en_US
+dynamic_entity.validation.required_data_is_missing_or_forbidden,"Einige erforderliche Daten fehlen oder die bereitgestellten Daten können nicht geändert werden. Bitte überprüfen Sie die Anfrage und versuchen Sie es erneut.",de_DE
```
Comment (asmarovydlo) [inline docs/scos/dev/feature-integration-guides/202304.0/glue-api/dynamic-data-api/dynamic-data-api-integration.md:205]: Customers should be able based on the response modify(fix) the data they send, and make the request one more time.
This message is entirely unclear. Please fix it here, and in the implementation.

**Review by asmarovydlo — CHANGES_REQUESTED** [review]

**Review by asmarovydlo — APPROVED** [review]

File: docs/scos/dev/glue-api-guides/202304.0/dynamic-data-api/how-to-guides/how-to-send-request-in-dynamic-data-api.md [hunk 9]
```diff
@@ -0,0 +1,456 @@
...
+        "postal_code_regex": null
+    }
+]
+```
+
+{% endinfo_block %}
+
+{% info_block infoBox %}
+
+Note, by default, API `GET` request returns up to 20 records.
```
Comment (AlexSlawinski) [inline docs/scos/dev/glue-api-guides/202304.0/dynamic-data-api/how-to-guides/how-to-send-request-in-dynamic-data-api.md:83]:
```suggestion
Note, by default the API `GET` request returns up to 20 records.
```

File: docs/scos/dev/glue-api-guides/202304.0/dynamic-data-api/how-to-guides/how-to-send-request-in-dynamic-data-api.md [hunk 10]
```diff
@@ -0,0 +1,456 @@
...
+  {
+    "iso2_code": "WA",
+    "iso3_code": "WWA",
+    "name": "FOO",
+    "id_country": 257
+  }
+]
+```
+
+The response payload includes all the specified fields from the request body, along with the id of the newly created entity.
```
Comment (AlexSlawinski) [inline docs/scos/dev/glue-api-guides/202304.0/dynamic-data-api/how-to-guides/how-to-send-request-in-dynamic-data-api.md:205]:
```suggestion
The response payload includes all the specified fields from the request body, along with the ID of the newly created entity.
```

File: docs/scos/dev/glue-api-guides/202304.0/dynamic-data-api/how-to-guides/how-to-send-request-in-dynamic-data-api.md [hunk 11]
```diff
@@ -0,0 +1,456 @@
...
+    "code": "505"
+  }
+]
+```
+
+{% endinfo_block %}
+
+{% info_block infoBox %}
+
+If the `id_country` is not provided you will get the following response:
```
Comment (AlexSlawinski) [inline docs/scos/dev/glue-api-guides/202304.0/dynamic-data-api/how-to-guides/how-to-send-request-in-dynamic-data-api.md:341]:
```suggestion
If `id_country` is not provided you will get the following response:
```

File: docs/scos/dev/glue-api-guides/202304.0/dynamic-data-api/how-to-guides/how-to-send-request-in-dynamic-data-api.md [hunk 12]
```diff
@@ -0,0 +1,456 @@
...
+[
+  {
+    "message": "Incomplete Request - missing identifier.",
+    "status": 400,
+    "code": "512"
+  }
+]
+```
+
+If the `id_country` is not found you will get the following response:
```
Comment (AlexSlawinski) [inline docs/scos/dev/glue-api-guides/202304.0/dynamic-data-api/how-to-guides/how-to-send-request-in-dynamic-data-api.md:353]:
```suggestion
If `id_country` is not found you will get the following response:
```

**Review by AlexSlawinski — APPROVED** [review]

**Review by AlexSlawinski — APPROVED** [review]

**Pull request merged**